back/engines/commercial/report_builder/app/services/report_builder/query_repository.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module ReportBuilder
+  class QueryRepository
+    GRAPH_RESOLVED_NAMES_CLASSES = {
+      'GenderWidget' => Queries::UsersByGender,
+      'ReactionsByTimeWidget' => Queries::ReactionsByTime
+    }.freeze
+
+    def data_by_graph(graph_resolved_name, props)
+      klass = GRAPH_RESOLVED_NAMES_CLASSES[graph_resolved_name]
+      return unless klass
+
+      run_query(klass.new.query(**props.to_h.symbolize_keys))
+    end
+
+    protected
+
+    def run_query(json_query)
+      # TODO: investigate why we need parameters and not HashWithIndifferentAccess
```

Comment:
Yes, it does. I just want to make the code simpler.

Reply:
@jamesspeake I started to use `HashWithIndifferentAccess` instead of `ActionController::Parameters` in this commit https://github.com/CitizenLabDotCo/citizenlab/pull/6309/commits/cc5e1eb89c3b2cce91c03313c588ce424cf501ab. Let me know if it can cause any issues.